packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -406,7 +407,12 @@ export const getRoutes = (locals) => {
     if (typeof routes === 'function') {
         _routes = routes()
     }
-    const allRoutes = [..._routes, {path: '*', component: Throw404}]
+    const allRoutes = [
+        // NOTE: this route needs to be above _routes, in case _routes has a fallback route of `path: '*'`
+        {path: '/__mrt/refresh', component: RefetchData},
```

Comment:
Do you think it would be easy to modify our current solution to include a full-page spinner instead of the current solution that renders the <App /> component's header and footer, with a page inside that loads a spinner, in case we get such a design request in the future?

Reply:
I guess it depends on what the full-page spinner would feel/look like. We can show the spinner on the Runtime Admin's side, instead of currently doing it in the storefront/iframe. But behind the scene, does the storefront still need to navigate away and back?